app/src/main/kotlin/github/com/rhacco/dota2androidapp/viewmodel/MatchesViewModel.kt

```diff
@@ -6,49 +6,85 @@ import android.arch.lifecycle.MediatorLiveData
 
     override fun onCleared() = mDisposables.clear()
 
-    fun getTopLiveGames() {
+    fun getLiveMatches() {
         mIsLoadingLiveData.value = true
         mDisposables.add(TopLiveGamesRepository.getTopLiveGames()
                 .subscribe(
                         { result ->
                             mIsLoadingLiveData.value = false
-                            // TODO string manipulation here i guess
-                            mTopLiveGamesQueryLiveData.value = result
+                            mLiveMatchesQueryLiveData.value = result
                         },
                         { error ->
                             mIsLoadingLiveData.value = false
                             Log.d(App.instance.getString(R.string.log_msg_debug),
-                                    "Failed to update top live games: " + error)
+                                    "Failed to fetch top live games: " + error)
                         }
                 ))
     }
 
-    fun getRealtimeStats(averageMMR: Int, serverSteamId: Long) {
+    fun getLiveMatchListData(averageMMR: Int, serverSteamId: Long) {
         mIsLoadingLiveData.value = true
         mDisposables.add(RealtimeStatsRepository.getRealtimeStats(serverSteamId)
                 .subscribe(
                         { result ->
                             mIsLoadingLiveData.value = false
-                            mRealtimeStatsQueryLiveData.value = Pair(averageMMR, result)
+                            val listItemData: LiveMatchesListAdapter.ListItemData =
+                                    LiveMatchesListAdapter.ListItemData()
+                            listItemData.mAverageMMR = averageMMR
+                            if (averageMMR < 1)
+                                listItemData.mTitle = "Tournament Match (Match ID " +
+                                        result.match.matchid + ")"
+                            else
+                                listItemData.mTitle = "Ranked Match (Average MMR " + averageMMR +
+                                        ", Match ID " + result.match.matchid + ")"
+                            if (result.teams.size == 2) {
+                                result.teams[0].players
+                                        .asSequence()
+                                        .map {
+                                            if (it.name.length > 20)
+                                                it.name.substring(0, 16) + "..."
+                                            else
+                                                it.name
+                                        }
+                                        .forEach {
+                                            listItemData.mPlayersRadiant =
+                                                    listItemData.mPlayersRadiant + "\n" + it
+                                        }
+                                result.teams[1].players
```

Comment:
DRY (Don't repeat yourself). You can do this with result.teams[i] looping.

Reply:
Fixed